api/v1/verticadb_webhook.go

```diff
@@ -1183,6 +1189,53 @@ func (v *VerticaDB) validateSandboxes(allErrs field.ErrorList) field.ErrorList {
 	return v.validateSubclustersInSandboxes(allErrs)
 }
 
+// validateProxyConfig validates if provided proxy info is correct
+func (v *VerticaDB) validateProxyConfig(allErrs field.ErrorList) field.ErrorList {
+	// check if we are using proxy deployments
+	if vmeta.UseVProxy(v.Annotations) {
+		// check if proxy set and image must be non-empty
+		if v.Spec.Proxy == nil {
+			err := field.Invalid(field.NewPath("spec").Child("proxy"),
+				"",
+				"proxy is not set")
+			allErrs = append(allErrs, err)
+		}
+		if v.Spec.Proxy != nil && v.Spec.Proxy.Image == "" {
+			err := field.Invalid(field.NewPath("spec").Child("proxy").Child("image"),
+				v.Spec.Proxy.Image,
+				"proxy.image cannot be empty")
+			allErrs = append(allErrs, err)
+		}
+
+		// check if replica is set and value is valid
+		for i := range v.Spec.Subclusters {
+			sc := &v.Spec.Subclusters[i]
+			// proxy replicas must be > 0(at all times)
+			if *sc.Proxy.Replicas <= 0 {
```

Comment:
You should check  sc.proxy is set `sc.proxy != nil ` and also check `sc.proxy.replicas != nil`

Reply:
Updated in https://github.com/vertica/vertica-kubernetes/pull/1000/commits/f6417faae82d7911e2a3479e2e6be3ddee1921ac